ts/Core/Axis/GridAxis.ts

```diff
@@ -1034,7 +1034,7 @@ class GridAxis {
                     'year',
                     null
                 ]]
-            }, userOptions);
+            } as Highcharts.AxisOptions, userOptions);
 
```

Comment:
I played a bit with the generic typing to make this working with `merge<Highcharts.AxisOptions>(true,…` and failed. My conclusion is here that TypeScript is right about the typing. With the forced type the problem only gets ignored, that you do in-place merge on a new created object which does not contain all defaults.
- Therefor the correct solution would be to simply do a regular merge without in-place flag here.
- Also `userOptions` and `gridAxisOptions` should be deep partial.
```ts
        e: { userOptions: DeepPartial<Highcharts.AxisOptions> }
    ): void {
        var options = this.options,
            userOptions = e.userOptions,
            gridAxisOptions: DeepPartial<Highcharts.AxisOptions>,
            gridOptions: GridAxis.Options = (
                (options && isObject(options.grid)) ? (options.grid as any) : {}
            );

        if (gridOptions.enabled === true) {

            // Merge the user options into default grid axis options so
            // that when a user option is set, it takes presedence.
            gridAxisOptions = merge<DeepPartial<Highcharts.AxisOptions>>({

                className: (
```